fenwick/src/main/java/org/opencadc/fenwick/InventoryHarvester.java

```diff
@@ -185,43 +194,108 @@ public InventoryHarvester(Map<String, Object> daoConfig, URI resourceID, Artifac
      */
     @Override
     public void run() {
+        boolean retry;
+        int retryCount = 1;
+        int sleepSeconds = INITIAL_SLEEP_TIMEOUT;
+
         while (true) {
             try {
                 final Subject subject = SSLUtil.createSubject(new File(CERTIFICATE_FILE_LOCATION));
-                Subject.doAs(subject, (PrivilegedExceptionAction<Void>) () -> {
-                    doit();
-                    return null;
+                int retries = retryCount;
+                int timeout = sleepSeconds;
+
+                retry = Subject.doAs(subject, (PrivilegedAction<Boolean>) () -> {
```

Comment:
The docs for PrivilegedAction say that run() does not throw checked exceptions, but I don't think checked vs unchecked is really the same as "something I can retry" vs "fatal". You will have to catch all checked exceptions inside run() even if some of those should be fatal. So the outer catch for fatal can only be unchecked (RuntimeException).

Reply:
Moved all exception checking into the run(), except for a RuntimeException wrapping the Subject.doAs.